### PR TITLE
ACS-8434: progress component

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- ACS-8434: lightweight `adf-progress` component that can render bars and spinners

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/progress/progress.component.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { MatButtonModule } from '@angular/material/button';
+
+export type ProgressVariant = 'bar' | 'spinner' | undefined;
+export type ProgressMode = ProgressBarMode | ProgressSpinnerMode;
+export type ProgressColor = ThemePalette;
+
+@Component({
+    selector: 'adf-progress',
+    standalone: true,
+    imports: [CommonModule, MatProgressBarModule, MatProgressSpinnerModule, MatButtonModule],
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/progress/progress.component.ts:32]: Is the import for MatButton needed?

Comment (DenysVuika) [reply lib/core/src/lib/progress/progress.component.ts]: good finding, removed the import, thanks

**Review by g-jaskowski — APPROVED** [review]

File: lib/core/src/lib/progress/progress.component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+adf-progress {
+    // default styles
```
Comment (DaryaBalvanovich) [inline lib/core/src/lib/progress/progress.component.scss:2]: do we need style file if it's empty?

**Review by nikita-web-ua — APPROVED** [review]

File: lib/core/src/lib/progress/progress.component.scss (hunk 2 above)
Comment (DenysVuika) [reply lib/core/src/lib/progress/progress.component.scss]: removed the file, thanks

**Review by DaryaBalvanovich — APPROVED** [review]

**Review by swapnil-verma-gl — APPROVED** [review]

**Review by kathrine0 — APPROVED** [review]

File: lib/core/src/lib/progress/progress.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            [value]="value"
+            [color]="color"
+            [attr.aria-label]="ariaLabel"
+            [attr.aria-hidden]="ariaHidden"
+            [attr.data-automation-id]="testId">
+        </mat-progress-bar>
+    </ng-container>
+
+    <ng-container *ngSwitchCase="'spinner'">
+        <mat-progress-spinner
```
Comment (DaryaBalvanovich) [inline lib/core/src/lib/progress/progress.component.html:14]: can we add `[diameter]` parameter to be able to change the size?
And may be it make sense to add any class or wrapper not to touch mat-selectors and change styles
For example, [here](https://github.com/Alfresco/alfresco-applications/pull/660/files#diff-a577b494dfb7e840dc813603dff14fd85f5c80cb5c704ce822e7ef8f5ec96f75R1) we have to use wrapper to make spinner in the center of the box

Comment (DenysVuika) [reply lib/core/src/lib/progress/progress.component.html:14]: We could probably investigate and do as new PRs and extensions. So far the diameter `looks` deprecated and might go away during our Angular upgrades

Screenshot 2024-07-23 at 9 51 44 AM

**Pull request merged**